### PR TITLE
[KYUUBI #410] Failed to start Spark 3.1 with error msg 'Cannot modify the value of a Spark config'

### PR DESCRIPTION
**Pull request by ulysses-you**
ulysses-you [Closes #440](https://github.com/yaooqinn/kyuubi/pull/440)    Test Plan Bug [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Fix the exception if we pass some unnecessary config to Spark.

It's hard to list the whole config which Spark disallowed, this patch just ingore the expcetion that Spark throw.

### _How was this patch tested?_
Pass the test `SessionSuite`

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @pan3793 @yaooqinn

**Review by pan3793 — APPROVED** [review]
LGTM

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -99,10 +99,11 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
...
 
   private def setModifiableConfig(spark: SparkSession, key: String, value: String): Unit = {
-    if (spark.conf.isModifiable(key)) {
+    try {
       spark.conf.set(key, value)
-    } else {
-      warn(s"Spark config $key is static and will be ignored")
+    } catch {
+      case _: AnalysisException =>
+        warn(s"Spark config $key is not modifiable during running, ignored it")
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:106]: nit: the original message of `AnalysisException` sounds more reasonable, the word `running` looks confusing

**Review by yaooqinn — APPROVED** [review]
anyway, lgtm. just a minor comment

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**